internal/rpc/jsonrpc/methods.go

```diff
@@ -3200,6 +3201,68 @@ func (s *Server) purchaseTicket(ctx context.Context, icmd interface{}) (interfac
 		dontSignTx = *cmd.DontSignTx
 	}
 
+	var csppServer string
+	var mixedAccount uint32
+	var mixedAccountBranch uint32
+	var mixedSplitAccount uint32
+	var changeAccount = account
+
+	if s.cfg.CSPPServer != "" {
+		csppServer = s.cfg.CSPPServer
+		mixedAccount, err = w.AccountNumber(ctx, s.cfg.MixAccount)
+		if err != nil {
+			return nil, rpcErrorf(dcrjson.ErrRPCInvalidParameter,
+				"CSPP Server set, but error on mixed account: %v", err)
+		}
+		mixedAccountBranch = s.cfg.MixBranch
+		if mixedAccountBranch != 0 && mixedAccountBranch != 1 {
+			return nil, rpcErrorf(dcrjson.ErrRPCInvalidParameter,
+				"MixedAccountBranch should be 0 or 1.")
+		}
+		_, err = w.AccountNumber(ctx, s.cfg.TicketSplitAccount)
+		if err != nil {
+			return nil, rpcErrorf(dcrjson.ErrRPCInvalidParameter,
+				"CSPP Server set, but error on mixedSplitAccount: %v", err)
+		}
+		_, err = w.AccountNumber(ctx, s.cfg.MixChangeAccount)
+		if err != nil {
+			return nil, rpcErrorf(dcrjson.ErrRPCInvalidParameter,
+				"CSPP Server set, but error on changeAccount: %v", err)
+		}
+	}
+
+	var vspHost string
+	var vspPubKey string
+	var vspClient *vsp.Client
+	if s.cfg.VSPHost != "" || s.cfg.VSPPubKey != "" {
+		vspHost = s.cfg.VSPHost
+		vspPubKey = s.cfg.VSPPubKey
+		if vspPubKey == "" {
+			return nil, rpcErrorf(dcrjson.ErrRPCInvalidParameter,
+				"vsp pubkey can not be null")
+		}
+		if vspHost == "" {
+			return nil, rpcErrorf(dcrjson.ErrRPCInvalidParameter,
+				"vsp host can not be null")
+		}
+		cfg := vsp.Config{
+			URL:    vspHost,
+			PubKey: vspPubKey,
+			Dialer: nil,
```

Comment:
I think this needs to be passed all the way from the main program entry (to `*Server`), otherwise purchasing via tor with `--torisolation` will not cause a new circuit to be used for each individual ticket purchase.

Similar to how `DialCSPPServer` is passed via the config of the `Server` struct.